metadata-ingestion/tests/unit/urns/test_urn.py

```diff
@@ -60,6 +66,20 @@ def test_urn_coercion() -> None:
     assert urn == Urn.from_string(urn.urn())
 
 
+def test_urns_in_init() -> None:
+    platform = DataPlatformUrn("abc")
+    assert platform.urn() == "urn:li:dataPlatform:abc"
+
+    dataset_urn = DatasetUrn(platform, "def", "PROD")
+    assert dataset_urn.urn() == "urn:li:dataset:(urn:li:dataPlatform:abc,def,PROD)"
+
+    schema_field = SchemaFieldUrn(dataset_urn, "foo")
+    assert (
+        schema_field.urn()
+        == "urn:li:schemaField:(urn:li:dataset:(urn:li:dataPlatform:abc,def,PROD),foo)"
+    )
```

Comment:
Any result different than this before looks like an error. So this seems more like a bug fix than a feat PR.

Reply:
All existing functionality is kept as is

It used be the case that things like SchemaFieldUrn(dataset, field) required dataset to be a string. But now it supports `DatasetUrn` as well. That's why I marked it as a feat.